File: benchmark/perl/count_set_bits.pl
use Time::HiRes qw(time);
my $start = time();
my $total = 0;
for my $n (1..100000) {
    my $x = $n;
    while ($x > 0) { $total++ if $x % 2 == 1; $x = int($x / 2); }
}
my $end = time();
my $ms = ($end - $start) * 1000;
print "Total: $total
";
printf "Time: %.0fms
", $ms;

File: benchmark/lua/count_set_bits.lua
local start = os.clock()
local total = 0
for n = 1, 100000 do
    local x = n + 0.0
    while x > 0.5 do
        if x % 2 >= 1 then
            total = total + 1
        end
        x = x / 2
    end
end
local finish = os.clock()
print("Total: " .. total)
print(string.format("Time: %.0fms", (finish - start) * 1000))

